client/scss/components/_drilldown.scss

```diff
@@ -60,10 +60,15 @@
   }
 }
 
+.w-drilldown__submenu .w-field__wrapper {
+  // Align the margin-top with the submenu’s back button
+  // and reduce the default margin-bottom.
+  margin-top: theme('spacing.2');
+  margin-bottom: theme('spacing.2');
+}
```

Comment:
Tweaked this a bit in dbf3c7c58626833205c435e824c4cbc464103ffc to prevent the focus outline from being cropped (and avoid the bottom margin from being too small)

## Before

<img width="320" alt="image" src="https://github.com/wagtail/wagtail/assets/6379424/085dd54f-b8f1-4f79-823e-08c9a6606e77">

## After

<img width="320" alt="image" src="https://github.com/wagtail/wagtail/assets/6379424/b9cadb1b-e51b-48b2-8505-028968626bbb">

I think we can use a bit more margin on the right side but I'm not too sure, so let's revisit it later